workflow/controller/workflowpod.go

```diff
@@ -138,6 +157,10 @@ func (woc *wfOperationCtx) createWorkflowPod(nodeName string, mainCtr apiv1.Cont
 		pod.ObjectMeta.Labels[common.LabelKeyControllerInstanceID] = woc.controller.Config.InstanceID
 	}
 
+	if common.GoogleSecretName != "" {
+		pod.Spec.Volumes = append(pod.Spec.Volumes, volumeGoogleSecret)
```

Comment:
We should only append this volume if we detect in the template, the user has a GCS input/output artifact. By always appending this volume and volume mount, this forces all workflow namespaces to have the same secret. For example, it should be possible to run the hello-world.yaml workflow in a brand new namespace (with no secrets).